test/extensions/tracers/xray/fuzz_test.cc

```diff
@@ -0,0 +1,22 @@
+#include "extensions/tracers/xray/util.h"
+
+#include "test/fuzz/fuzz_runner.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace XRay {
+
+DEFINE_FUZZER(const uint8_t* buf, size_t len) {
+  absl::string_view sv(reinterpret_cast<const char*>(buf), len);
+  wildcardMatch("*", sv);
+  wildcardMatch("?", sv);
+  wildcardMatch("*??", sv);
+  wildcardMatch("??*", sv);
+  wildcardMatch(sv, "hello, world");
```

Comment:
Incidentally, this would also be a perfect application of differential fuzzing (CC @asraa). You could use fuzzing to effectively do a correctness validation (not just a crash test) by comparing the result from `wildcardMatch` to the slower/more laborious RE2 based match (where you just substitute `.*` for every `*`, `.` for every `?` and escape the rest as needed). I'd say this is optional, but you might want to leave a TODO here for folks interested in picking up fuzzing tasks.